File: code_samples/lua/networking_http_client.lua
-- Networking: Http Client
-- AI/ML Training Sample

HttpClient = {}
HttpClient.__ index = HttpClient

function HttpClient:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function HttpClient:process(input)
    self.data = input
end

function HttpClient:getData()
    return self.data
end

function HttpClient:validate()
    return #self.data > 0
end

-- Example usage
local instance = HttpClient:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))

File: code_samples/dart/networking_http_client.dart
// Networking: Http Client
// AI/ML Training Sample

class HttpClient {
  String _data = '';
  
  void process(String input) {
    _data = input;
  }
  
  String getData() => _data;
  
  bool validate() => _data.isNotEmpty;
}

void main() {
  final instance = HttpClient();
  instance.process('example');
  print('Data: ${instance.getData()}');
  print('Valid: ${instance.validate()}');
}
